src/mlpack/methods/cf/normalization/no_normalization.hpp

```diff
@@ -0,0 +1,79 @@
+/**
+ * @file no_normalization.hpp
+ * @author Wenhao Huang
+ *
+ * This class performs no normalization. It is used as default type of
+ * normalization for CF class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_CF_NORMALIZATION_NO_NORMALIZATION_HPP
+#define MLPACK_METHODS_CF_NORMALIZATION_NO_NORMALIZATION_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace cf {
+
+/**
+ * This normalization class doesn't perform any normalization. It is the default
+ * normalization type for CF class.
+ */
+class NoNormalization
+{
+ public:
+  // Empty constructor.
+  NoNormalization() { }
+
+  /**
+   * Do nothing.
+   *
+   * @param data Input dataset in the form of coordinate list.
+   */
+  inline void Normalize(const arma::mat& /* data */) const { }
+
+  /**
+   * Do nothing.
+   *
+   * @param cleanedData Sparse matrix data.
+   */
+  inline void Normalize(const arma::sp_mat& /* cleanedData */) const { }
+
+  /**
+   * Do nothing.
+   *
+   * @param user User ID.
+   * @param item Item ID.
+   * @param rating Computed rating before denormalization.
+   */
+  inline double Denormalize(const size_t /* user */,
+                            const size_t /* item */,
+                            const double rating) const
+  {
+    return rating;
+  }
+
+  /**
+   * Do nothing.
+   *
+   * @param combinations User/Item combinations.
+   * @param predictions Predicted ratings for each user/item combination.
+   */
+  inline void Denormalize(const arma::Mat<size_t>& /* combinations */,
+                          const arma::vec& /* predictions */) const
+  { }
+
+  /**
+   * Serialization.
+   */
+  template<typename Archive>
+  void serialize(Archive& /* ar */, const unsigned int /* version */) { }
```

Comment:
No need to provide the function if it's empty (we use enable_if to detect if the function is available).

Reply:
```
/usr/local/include/boost/serialization/access.hpp:116:11: error: no member named 'serialize' in
      'mlpack::cf::NoNormalization'
        t.serialize(ar, file_version);
```
It generated a compilation error after I removed the `serialize()` function in `NoNormalization`. Any idea why this error occurs?

Reply:
Ah, I guess since you call the serialize method inside the ``CombinedNormalization`` class, we can't just remove the function. I guess in this case let's leave the empty function for now.

Reply:
Right I forgot about that, thanks!